librfn/libopencm3/console_cdcacm.c

```diff
@@ -233,28 +233,26 @@ static struct output_task output_task = {
 #if defined(STM32F1)
 static void usb_hwinit(void)
 {
-	uint32_t t;
-
 	rcc_periph_clock_enable(RCC_GPIOA);
-	rcc_periph_clock_enable(RCC_GPIOB);
+	rcc_periph_clock_enable(RCC_AFIO);
 
-	/* lower hotplug and leave enough time for the host to notice */
-	gpio_set_mode(GPIOA, GPIO_MODE_OUTPUT_2_MHZ, GPIO_CNF_OUTPUT_PUSHPULL,
-		      GPIO11 | GPIO12);
-	gpio_clear(GPIOA, GPIO11 | GPIO12);
-	t = time_now() + 10000;
-	while (cyclecmp32(time_now(), t) < 0)
-		;
```

Comment:
Without the 20ms delay before asserting hot plug, you are likely to have problems handing over from a DFU bootloader to the main application (host might not notice the device has changed protocol).